articles/architecture-scenarios/b2c/user-provisioning.md

```diff
@@ -0,0 +1,83 @@
+---
+title: Consumer Identity and Access Management User Provisioning
+description: Understand how user provisioning works in your CIAM implementation. 
+toc: true
+topics:
+    - b2c
+    - ciam
+    - user-migration
+    - identity-store
+contentType: concept
+useCase:
+  - user-provisioning
+---
+# Consumer Identity and Access Management User Provisioning
+
+User provisioning defines how users will be added to your system. This part of the implementation determines where you will store your users' profile information, how you will migrate existing users, and how your users will sign up. 
+
+## Prerequisites
+
+* Tenant Architecture
+
+## Design considerations
+
+It is important to decide early in your implementation planning how you want users added. The decisions you make influence many other implementation choices.
+
+The factors to consider are: 
+
+1. Where to store your user data, in your own identity data store or in Auth's data store?
+2. If you have legacy user data, do you want to migrate it to Auth0?
+3. Do you want your users to sign up with their existing accounts like Facebook or Google? 
+
+## Identity data storage
+
+During sign up, Auth0 creates a [user profile](/users/concepts/overview-user-profile) that contains information about the user’s account. You can choose to store the user information in Auth0's data store or your own. 
+
+### Use your own identity data store
+
+If you want to use your own identity store, perhaps because you’ve got applications which you aren’t ready to migrate or can’t be migrated, use Auth0’s [Database Connections](/connections/database/custom-db) to proxy an existing identity store. 
+
+### Use Auth0 as the identity data store
+
+Auth0 provides identity data storage to help you manage the burden of storing user credentials safely and securely. If you have a legacy identity store that you would like Auth0 to host, you can use Auth0’s user migration capabilities. 
+
+## User migration
+
+If you decide to use Auth0 instead of your own identity store, you can [migrate users](/users/concepts/overview-user-migration) with bulk migration or progressively with automatic migration. You can also choose a two-phased approach to user migration, using automatic migration first, then using bulk migration for the users that remain. For other scenarios, see [User Migration Scenarios](/users/references/user-migration-scenarios).
```

Comment:
If you decide to use Auth0 as your data store, you can migrate users in one of two ways:

* With bulk migration to bring in all of your data at once
* With progressive automatic migration to bring in your data over time

Keep in mind that there are multiple approaches to user migration - you are not obligated to choose only one option. For example, you might opt for a two-phased approach to migration, beginning with automatic migration and following it up with bulk migration for the remaining users. See User Migration Scenarios for additional options.